ciri/api/api.c

```diff
@@ -144,9 +95,37 @@ retcode_t iota_api_find_transactions(find_transactions_req_t const *const req,
   return RC_OK;
 }
 
-retcode_t iota_api_get_trytes(get_trytes_req_t const *const req,
+retcode_t iota_api_get_trytes(iota_api_t const *const api,
+                              get_trytes_req_t const *const req,
                               get_trytes_res_t *const res) {
-  return RC_OK;
+  retcode_t ret = RC_OK;
+  flex_hash_array_t *iter = NULL;
+  flex_trit_t tx_trits[FLEX_TRIT_SIZE_8019];
+  tryte_t tx_trytes[NUM_TRYTES_SERIALIZED_TRANSACTION + 1];
+  DECLARE_PACK_SINGLE_TX(tx, txp, pack);
+
+  if (flex_hash_array_count(req->hashes) > api->limits.max_get_trytes) {
+    return RC_API_MAX_GET_TRYTES;
+  }
+  LL_FOREACH(req->hashes, iter) {
+    hash_pack_reset(&pack);
+    if ((ret = iota_tangle_transaction_load(api->tangle, TRANSACTION_FIELD_HASH,
```

Comment:
Could you add a comment here that implies that we need to consider concurrency here?

Reply:
Due to statements ?

Reply:
Yeah...